pkg/controller/jobs/pod/pod_controller.go

```diff
@@ -737,9 +737,9 @@ func (p *Pod) notRunnableNorSucceededPods() []corev1.Pod {
 }
 
 // isPodRunnableOrSucceeded returns whether the Pod can eventually run, is Running or Succeeded.
-// A Pod cannot run if it's gated and has a deletionTimestamp.
+// A Pod cannot run if it's gated or has no node assignment while having a deletionTimestamp.
 func isPodRunnableOrSucceeded(p *corev1.Pod) bool {
-	if p.DeletionTimestamp != nil && len(p.Spec.SchedulingGates) > 0 {
+	if p.DeletionTimestamp != nil && len(p.Spec.NodeName) == 0 {
```

Comment:
Please rename the PR/commits to something more precise. The current title suggests that you will be marking all unschedulable pods (including these waiting for ClusterAutoscaler) as inactive, not only these that are already deleted and doesn't have a node name assigned.

Reply:
done